src/localFunction.ts

```diff
@@ -33,25 +35,122 @@ export default class LocalFunction {
+      method: (path: string, body?: any, opts?: ClientVerbOptions) => Promise<ClientResponse<any>>
+    ): verbFn => {
+      return async (pathOrOptions?: string | HttpsOptions, options?: HttpsOptions) => {
+        const { path, opts } = this.extractArgs(pathOrOptions, options);
+        await method(path, opts.body, toClientVerbOptions(opts))
+          .then((res) => {
+            this.requestCallBack(undefined, res, res.body);
+          })
+          .catch((err) => {
+            this.requestCallBack(err);
+          });
+        return HTTPS_SENTINEL;
+      };
+    };
+    const verbWithoutReqBodyFactory = (
+      method: (path: string, options?: ClientVerbOptions) => Promise<ClientResponse<any>>
+    ): verbFn => {
+      return async (pathOrOptions?: string | HttpsOptions, options?: HttpsOptions) => {
+        const { path, opts } = this.extractArgs(pathOrOptions, options);
+        await method(path, toClientVerbOptions(opts))
+          .then((res) => {
+            this.requestCallBack(undefined, res, res.body);
+          })
+          .catch((err) => {
+            this.requestCallBack(err);
+          });
+        return HTTPS_SENTINEL;
+      };
+    };
+
+    const shim = verbWithReqBodyFactory((path: string, json?: any, opts?: ClientVerbOptions) => {
+      const req = Object.assign(opts || {}, {
+        path: path,
+        body: json,
+        method: opts?.method || "GET",
+      });
+      return callClient.request(req);
     });
+    const verbs: verbMethods = {
+      post: verbWithReqBodyFactory((path: string, json?: any, opts?: ClientVerbOptions) =>
+        callClient.post(path, json, opts)
+      ),
+      put: verbWithReqBodyFactory((path: string, json?: any, opts?: ClientVerbOptions) =>
+        callClient.put(path, json, opts)
+      ),
+      patch: verbWithReqBodyFactory((path: string, json?: any, opts?: ClientVerbOptions) =>
+        callClient.patch(path, json, opts)
+      ),
+      get: verbWithoutReqBodyFactory((path: string, opts?: ClientVerbOptions) =>
+        callClient.get(path, opts)
+      ),
+      del: verbWithoutReqBodyFactory((path: string, opts?: ClientVerbOptions) =>
+        callClient.delete(path, opts)
+      ),
+      delete: verbWithoutReqBodyFactory((path: string, opts?: ClientVerbOptions) =>
+        callClient.delete(path, opts)
+      ),
+      options: verbWithoutReqBodyFactory((path: string, opts?: ClientVerbOptions) =>
+        callClient.options(path, opts)
+      ),
+    };
+    return Object.assign(shim, verbs);
+  }
+
+  private extractArgs(
+    pathOrOptions?: string | HttpsOptions,
+    options?: HttpsOptions
+  ): { path: string; opts: HttpsOptions } {
+    let path = "/";
+    let opts: HttpsOptions = {};
+    // Just path
+    if (typeof pathOrOptions === "string" && !options) {
+      path = pathOrOptions;
+    } // error case, 2 options
+    else if (!!pathOrOptions && !!options && typeof pathOrOptions !== "string") {
+      throw new Error(
```

Comment:
I had hard time evaluating this series of if statement in my head.

wdyt about this implementation?

```js
  // Case: No arguments provided
  if (!pathOrOptions && !options) {
    return { path: "/", opts: {} };
  }

  // Case: pathOrOptions is provided as a string
  if (typeof pathOrOptions === "string") {
    return { path: pathOrOptions, opts: options || {} };
  }

  // Case: pathOrOptions is an object (HttpsOptions), and options is not provided
  if (typeof pathOrOptions !== "string" && !options) {
    return { path: "/", opts: pathOrOptions };
  }

  // Error case: Invalid combination of arguments
  if (typeof pathOrOptions !== "string" || !options) {
    throw new Error(
      `Invalid argument combination: Expected a string and/or HttpsOptions, got ${typeof pathOrOptions} and ${typeof options}`
    );
  }

  // Default return, though this point should not be reached
  return { path: "/", opts: {} };
```

Reply:
Oooh, that is much more readable, thanks!